monix-reactive/shared/src/main/scala/monix/reactive/internal/builders/InputStreamObservable.scala

```diff
@@ -38,10 +38,12 @@ private[reactive] final class InputStreamObservable(in: InputStream, chunkSize:
   self =>
 
   private[this] val wasSubscribed = Atomic(false)
+  private[this] val fallbackChunkSize = 4096
```

Comment:
Yeah, it hangs, since the read will always be zero, the process will be stuck in `fastLoop`, so we should somehow avoid a zero-sized buffer.

The `fallbackChunkSize` I used is the same as what the default value in `Observable.fromInputStream` or `Observable.fromCharsReader`. 

I like the idea of signaling an error instead, because it is a bad input anyway. Let us hear what the others have to say.

Reply:
👍 signal error.

Reply:
Thanks for the advice @alexandru. @Avasil , I updated the PR based on the advice, and I wrote some comments on the file changes to explain some implementation decisions.